core/server/api_container/server/startosis_engine/kurtosis_types/service_config/ready_conditions.go

```diff
@@ -0,0 +1,213 @@
+	"go.starlark.net/starlark"
+	"time"
+)
+
+const (
+	ReadyConditionsTypeName = "ReadyConditions"
+
+	RecipeAttr    = "recipe"
+	FieldAttr     = "field"
+	AssertionAttr = "assertion"
+	TargetAttr    = "target_value"
+	IntervalAttr  = "interval"
+	TimeoutAttr   = "timeout"
+
+	defaultInterval = 1 * time.Second
+	defaultTimeout  = 15 * time.Minute //TODO we could move these two to the service helpers method
+)
+
+func NewReadyConditionsType() *kurtosis_type_constructor.KurtosisTypeConstructor {
+	return &kurtosis_type_constructor.KurtosisTypeConstructor{
+		KurtosisBaseBuiltin: &kurtosis_starlark_framework.KurtosisBaseBuiltin{
+			Name: ReadyConditionsTypeName,
+			Arguments: []*builtin_argument.BuiltinArgument{
+				{
+					Name:              RecipeAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.Value],
+					Validator:         nil,
+				},
+				{
+					Name:              FieldAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+				{
+					Name:              AssertionAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         assert.ValidateAssertionToken,
+				},
+				{
+					Name:              TargetAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.Comparable],
+					Validator:         nil,
+				},
+				{
+					Name:              IntervalAttr,
+					IsOptional:        true,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+				{
+					Name:              TimeoutAttr,
+					IsOptional:        true,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+			},
+		},
+		Instantiate: instantiateReadyConditions,
+	}
+}
+
+func instantiateReadyConditions(arguments *builtin_argument.ArgumentValuesSet) (kurtosis_type_constructor.KurtosisValueType, *startosis_errors.InterpretationError) {
+	kurtosisValueType, err := kurtosis_type_constructor.CreateKurtosisStarlarkTypeDefault(ReadyConditionsTypeName, arguments)
+	if err != nil {
+		return nil, err
+	}
+	return &ReadyConditions{
+		KurtosisValueTypeDefault: kurtosisValueType,
+		recipe:                   nil,
+		field:                    "",
+		assertion:                "",
+		target:                   starlark.String(""),
+		interval:                 defaultInterval,
+		timeout:                  defaultTimeout,
+	}, nil
+}
+
+// ReadyConditions is a starlark.Value that holds all the information needed for ensuring service readiness
+type ReadyConditions struct {
+	*kurtosis_type_constructor.KurtosisValueTypeDefault
+	recipe    recipe.Recipe
+	field     string
+	assertion string
+	target    starlark.Comparable
+	interval  time.Duration
+	timeout   time.Duration
```

Comment:
Out of curiosity why do you need to store those fields? They are already stored inside the KurtosisValueTypeDefault and you can extract them on demand, maybe?

Reply:
I wanted to do it this way because it's necessary to validate all of them at the same time during the interpretation and then I save them in the object because of this method:
<img width="852" alt="image" src="https://user-images.githubusercontent.com/29951623/225017668-61d379b8-7751-49e2-ba3e-cc3bcffc97aa.png">

That will be called then here:
<img width="620" alt="image" src="https://user-images.githubusercontent.com/29951623/225018133-4a1faa58-d22e-4f83-9b83-b7f77380c65b.png">

<img width="931" alt="image" src="https://user-images.githubusercontent.com/29951623/225018422-3cbc40b5-7890-4106-94c7-73a23d9d60d1.png">


And the object it's needed here:
<img width="562" alt="image" src="https://user-images.githubusercontent.com/29951623/225018252-6f9389b2-1f34-46cf-a7bb-4c073cf97342.png">